docs/manual/common/concepts/ServiceDiscovery.md

```diff
@@ -0,0 +1,30 @@
+# Service Discovery
+
+Microservice may scale up and down and also must stay mobile. In those scenarios, a Service Registry (also referred to [[Service Locator|ServiceLocator]]) is required. It contains an updated list of available microservices with a list of reachable host and ip's where each microservive can be found.
+
+In order to register every microservice in the Service Locator there are two alternatives: self-registration and [3rd Party Registration](http://microservices.io/patterns/3rd-party-registration.html). Lagom uses 3rd Party Registration and provides an implementation of a Service Locator for the [[Developer Environment|DevEnvironment]]. When booting a Lagom microservice the registrator will register the name of the microservice, the URL and the names of the Service Descriptor on the Service Locator so that it can be located. 
+
+Lagom supports deploying many Service Descriptors inside a single microservice. That is not against the recommendation that a [[microservice should only do one thing|Microservices#does-this-service-do-only-one-thing-]] and do it well. Following are the specifications of a `Greetings` microservice that contains two Service Descriptors (`welcome` and `farewell`) : 
```

Comment:
I think you need to justify why it's not against the recommendation - eg, it's often useful to provide additional meta services that provide metrics or back office admi.nistration capabilities, and also there may be reasons to split the one function into multiple logical services, for future proofing for example.